core/services/workflows/engine_test.go

```diff
@@ -106,11 +128,14 @@ func newTestEngine(t *testing.T, reg *coreCap.Registry, spec string, opts ...fun
 			executionFinished <- weid
 		},
 		clock: clock,
-		Store: store.NewDBStore(pgtest.NewSqlxDB(t), clock),
 	}
 	for _, o := range opts {
 		o(&cfg)
 	}
+	// We use the cfg clock incase they override it
+	if cfg.Store == nil {
+		cfg.Store = newTestDBStore(t, cfg.clock)
```

Comment:
@HenryNguyen5 This means we won't be able to override the store with options; I think we do that in a few tests.

It's actually not a big deal because the store hits the database so the tests won't fail, but in that case could we clean up the tests that do this so they don't override the store when calling newTestEngine?

Reply:
We can though? Unless I misunderstand you. I only set `cfg.Store` if one of the option handlers hasn't set the store

Reply:
I misread the equality sign 🤦